autostarter/src/main/java/com/judemanutd/autostarter/AutoStartPermissionHelper.kt

```diff
@@ -86,267 +86,210 @@ class AutoStartPermissionHelper private constructor() {
     private val BRAND_SAMSUNG = "samsung"
     private val PACKAGE_SAMSUNG_MAIN = "com.samsung.android.lool"
     private val PACKAGE_SAMSUNG_COMPONENT = "com.samsung.android.sm.ui.battery.BatteryActivity"
-    private val PACKAGE_SAMSUNG_COMPONENT_2 = "com.samsung.android.sm.battery.ui.BatteryActivity"
+    private val PACKAGE_SAMSUNG_COMPONENT_2 = "com.samsung.android.sm.battery.ui.usage.CheckableAppListActivity"
+    private val PACKAGE_SAMSUNG_COMPONENT_3 = "com.samsung.android.sm.battery.ui.BatteryActivity"
 
     /***
      * One plus
      */
     private val BRAND_ONE_PLUS = "oneplus"
     private val PACKAGE_ONE_PLUS_MAIN = "com.oneplus.security"
     private val PACKAGE_ONE_PLUS_COMPONENT = "com.oneplus.security.chainlaunch.view.ChainLaunchAppListActivity"
+    private val PACKAGE_ONE_PLUS_ACTION = "com.android.settings.action.BACKGROUND_OPTIMIZE"
 
     private val PACKAGES_TO_CHECK_FOR_PERMISSION = listOf(PACKAGE_ASUS_MAIN, PACKAGE_XIAOMI_MAIN, PACKAGE_LETV_MAIN, PACKAGE_HONOR_MAIN, PACKAGE_OPPO_MAIN,
             PACKAGE_OPPO_FALLBACK, PACKAGE_VIVO_MAIN, PACKAGE_VIVO_FALLBACK, PACKAGE_NOKIA_MAIN, PACKAGE_HUAWEI_MAIN, PACKAGE_SAMSUNG_MAIN, PACKAGE_ONE_PLUS_MAIN)
 
-    fun getAutoStartPermission(context: Context): Boolean {
+    /**
+     * It will attempt to open the specific manufacturer settings screen with the autostart permission
+     * If [open] is changed to false it will just check the screen existence
+     *
+     * @param context
+     * @param open, if true it will attempt to open the activity, otherwise it will just check its existence
+     * @param newTask, if true when the activity is attempted to be opened it will add FLAG_ACTIVITY_NEW_TASK to the intent
+     * @return true if the activity was opened or is confirmed that it exists (depending on [open]]), false otherwise
+     */
+    fun getAutoStartPermission(context: Context, open: Boolean = true, newTask: Boolean = false): Boolean {
 
-        when (Build.BRAND.toLowerCase(Locale.getDefault())) {
+        when (Build.BRAND.toLowerCase(Locale.ROOT)) {
```

Comment:
@Prometheus3k I think you guys will need to sort out which one actually work here because I don't have the resources to actually verify this. @awoisoak for context, there is a [PR](https://github.com/judemanutd/AutoStarter/pull/68/files#diff-26f6042147d7afa3e4fdbe58707a0973591ba0a88db20e8c216d4ae374b82114R103) that updates this very line.

Reply:
My PR is a restoration of @alperenbabagil Original work that was approved previously. 
https://github.com/judemanutd/AutoStarter/pull/58

Reply:
Thanks, I just tested with `Locale.ROOT` as is the approach we are following in similar scenarios in our side.